src/lp/empty_row.jl

```diff
@@ -1,21 +1,73 @@
-# TODO: this is redundant with forcing constraint checks
-#   => an empty row is automatically redundant or infeasible
+@doc raw"""
+    RemoveEmptyRow <: AbstractRule
 
+Eliminate a single row if it is empty, i.e. all variable coefficients are zero.
+In particular, consider row $i$ given as
+```math
+l^{r}_{i} ≤ \sum_{j} a_{i, j} x_{j} ≤ u^{r}_{i}.
+```
+We eliminate row $i$ if, for each variable index $j$,
+```math
+| a_{i,j} | ≤ ϵ
+```
+for prescribed tolerance $ϵ$.
+
+## Presolve
+
+If row $i$ is empty, the corresponding constraint is completely removed from
+the problem.
+
+## Infeasibility checks
+
+The problem is infeasible if row $i$ is empty and either 1) its
+lower bound is positive ($l^{r}_{i} > ϵ$) or 2) its upper bound is
+negative ($u^{r}_{i} < -ϵ$). In these two cases, a Farkas infeasibility
+certificate is given by setting $y$ to all zeros except for either
+$y\_lower_{i} = 1$ or $y\_upper_{i} = 1$, respectively.
```

Comment:
Documenter renders this in a single 3-line paragraph, which I find hard to read at first glance.
I'd suggest something like

```
For a prescribed tolerance $ϵ ≥ 0$, the problem is infeasible if either
1. row $i$ is empty and its lower bound is positive, i.e., $l^{r}_{i} > ϵ$, or
2. row $i$ is empty and its upper bound is negative, i.e., $u^{r}_{i} < -ϵ$.

The corresponding Farkas infeasibilities certificates are
1.  $y^{l} = e_{i}$, $y^{u} = 0$, $z^{l} = z^{u} = 0$,
2.  $y^{u} = e_{i}$, $y^{l} = 0$, $z^{l} = z^{u} = 0$,

where $e_{i}$ is the $i$-th vector of the canonical basis.
```
(it needs two spaces between the bullet point and the `$` to render correctly)